tiledb/sm/filter/compression_filter.cc

```diff
@@ -396,6 +435,144 @@ Status CompressionFilter::decompress_part(
   return st;
 }
 
+std::vector<std::string_view> CompressionFilter::create_input_view(
+    const FilterBuffer& input, Tile* const offsets_tile) {
+  auto input_buf = static_cast<const char*>(input.buffers()[0].data());
+  auto offsets_data = static_cast<uint64_t*>(offsets_tile->data());
+  auto offsets_size = offsets_tile->size() / constants::cell_var_offset_size;
+  std::vector<std::string_view> input_view(offsets_size);
+
+  size_t i = 0;
+  for (i = 0; i < offsets_size - 1; i++) {
+    input_view[i] = std::string_view(
+        input_buf + offsets_data[i], offsets_data[i + 1] - offsets_data[i]);
+  }
+
+  // special case for the last string
+  input_view[i] = std::string_view(
+      input_buf + offsets_data[i], input.size() - offsets_data[i]);
+
+  return input_view;
+}
+
+Status CompressionFilter::compress_var_string_coords(
+    const FilterBuffer& input,
+    Tile* const offsets_tile,
+    FilterBuffer& output,
+    FilterBuffer& output_metadata) const {
+  if (input.num_buffers() != 1) {
+    return LOG_STATUS(
+        Status_FilterError("Var-sized string input has to be in single "
+                           "buffer format to be compressed with RLE"));
+  }
+
+  // Construct string view of input
+  auto input_view = create_input_view(input, offsets_tile);
+
+  // Estimate and allocate output size
+  /*
+   * TBD: another option instead of traversing the whole input to get the
```

Comment:
I think this would be a good potential followup. I think we initially talked about the two-pass method as the "safe and easy" first implementation.